File: benchmarks/tak/tak.cs
// Tak Benchmark - C#
// Build: dotnet build -c Release
// Run: dotnet run -c Release

class Program {
    static int Tak(int x, int y, int z) {
        if (y >= x) return z;
        return Tak(Tak(x - 1, y, z), Tak(y - 1, z, x), Tak(z - 1, x, y));
    }

    static void Main() {
        int result = Tak(24, 16, 8);
        Console.WriteLine(result);
    }
}

File: benchmarks/tak/tak.cpp
// Tak Benchmark - C++
// Compile: clang++ -O3 -o tak_cpp tak.cpp
#include <iostream>

int tak(int x, int y, int z) {
    if (y >= x) return z;
    return tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y));
}

int main() {
    int result = tak(24, 16, 8);
    std::cout << result << std::endl;
    return 0;
}
